commands/project/dev/index.ts

```diff
@@ -21,36 +19,54 @@ import {
   logProfileHeader,
   exitIfUsingProfiles,
 } from '../../../lib/projectProfiles';
+import { commands } from '../../../lang/en';
+import { getAccountId } from '@hubspot/local-dev-lib/config';
 
 const command = 'dev';
-const describe = uiBetaTag(
-  i18n(`commands.project.subcommands.dev.describe`),
-  false
-);
+const describe = uiBetaTag(commands.project.dev.describe, false);
 
 async function handler(
   args: ArgumentsCamelCase<ProjectDevArgs>
 ): Promise<void> {
-  const { derivedAccountId, providedAccountId } = args;
+  const {
+    derivedAccountId,
+    providedAccountId,
+    testingAccount,
+    projectAccount,
+  } = args;
 
   const { projectConfig, projectDir } = await getProjectConfig();
   validateProjectConfig(projectConfig, projectDir);
 
+  const useV3 = useV3Api(projectConfig.platformVersion);
+
   if (!projectDir) {
-    logger.error(
-      i18n(`commands.project.subcommands.dev.errors.noProjectConfig`, {
-        accountId: derivedAccountId,
-        authCommand: uiCommandReference('hs auth'),
-      })
-    );
+    uiLogger.error(commands.project.dev.errors.noProjectConfig);
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  // If either targetTestingAccount or targetProjectAccount is provided, the other must be provided
+  if (
+    (testingAccount && !projectAccount) ||
+    (projectAccount && !testingAccount)
+  ) {
+    uiLogger.error(commands.project.dev.errors.invalidAccountFlags);
     process.exit(EXIT_CODES.ERROR);
   }
 
-  let targetAccountId = providedAccountId;
+  // Legacy projects do not support targetTestingAccount and targetProjectAccount
+  if (testingAccount && projectAccount && !useV3) {
+    uiLogger.error(commands.project.dev.errors.unsupportedAccountFlags);
+    process.exit(EXIT_CODES.ERROR);
+  }
+
+  let targetProjectAccountId =
+    (providedAccountId && derivedAccountId) ||
```

Comment:
iirc `derivedAccountId` will always exist if `providedAccountId` exists

Reply:
That's what I thought initially, but `providedAccountId` is kind of a misnomer - its whatever value the user passed in to the account flag, so if it's a name, we don't convert it to and id. If it exists, then `derivedAccountId` will be the provided account id converted to an id if necessary.

There are a LOT of unnecessary conversions between account name / id / full config object so trying to cut back on those where possible

Reply:
> If it exists, then derivedAccountId will be the provided account id converted to an id if necessary.

If that's true, then isn't checking `providedAccountId` the same as checking `providedAccountId && derivedAccountId`?